dotenv/mod.ts

```diff
@@ -1,191 +1,336 @@
 // Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
-/**
- * Dotenv config
- * @module
- */
 
-import { difference, removeEmptyValues } from "./util.ts";
+import {
+  DenoEnv,
+  Env,
+  EnvObject,
+  optionalReadTextFile,
+  optionalReadTextFileSync,
+  setDenoEnv,
+} from "./_util.ts";
 
-export interface DotenvConfig {
-  [key: string]: string;
-}
+export type { DenoEnv, Env, EnvObject };
 
-export interface ConfigOptions {
-  path?: string;
-  export?: boolean;
-  safe?: boolean;
-  example?: string;
+export interface VerifyOptions {
   allowEmptyValues?: boolean;
-  defaults?: string;
+  defaultsEnv?: Env;
 }
+/**
+ * @param object EnvObject that should be verified.
+ * @param options verification options
+ * @returns true if valid, else throws an error
+ * ```ts
+ * import { verify } from "https://deno.land/std@$STD_VERSION/dotenv/mod.ts";
+ *
+ * const object = { env: { GREETING: "hello world" }, exports: [] };
+ * const exampleEnv = { GREETING: "someValue" };
+ * const options = { allowEmptyValues: true };
+ * verify(object, exampleEnv, options);
+ * ```
+ */
+export function verify(
```

Comment:
`verify` looks like an internal utility to me. Should this be exported? (What is verified by this is not documented in jsdoc or README. I guess the users don't understand what's done by this.)